app/controllers/courses_controller.rb

```diff
@@ -180,7 +180,7 @@ def save_unlocks
     groups = group_params
     groups.each do |name, conditions|
       array = []
-      conditions.each { |k, v| array << v }
+      conditions.each { |k, v| array << v unless v.blank? }
```

Comment:
I'm wondering if we have better way for doing this; `.flatten!` and `.uniq!` might be helpful.